schema/bool_test.go

```diff
@@ -186,17 +186,13 @@ func TestBoolSerializationCycle(t *testing.T) {
 }
 
 func TestBoolJSONMarshal(t *testing.T) {
-	j, err := json.Marshal(schema.NewBoolSchema())
-	if err != nil {
-		t.Fatal(err)
-	}
+	boolType := schema.NewBoolSchema()
+	j, err := json.Marshal(&boolType)
+	assert.NoError(t, err)
 	if string(j) != "{}" {
 		t.Fatalf("Unexpected JSON output: %s", j)
 	}
-	boolType := schema.NewBoolSchema()
-	if err := json.Unmarshal(j, &boolType); err != nil {
-		t.Fatal(err)
-	}
+	assert.NoError(t, json.Unmarshal(j, &boolType))
```

Comment:
I don't think so in this case, because there are no variations of BoolSchema. That is due to there being no properties within that type. So no matter what it will be equal.

Reply:
Your statement assumes that the underlying implementation of `BoolSchema` will never change.  We want the tests to be able to verify the CUT, even if details of its implementation are changed.

But, as we discussed in Slack, all the changes to this module are out of scope for this PR, so I'm not going to hold it up for this.